File: backend/scripts/seed.py
"""
Seed script — creates default admin and teacher users, and default IT courses.
Run once on first deploy: python scripts/seed.py
"""
import sys, os
sys.path.insert(0, os.path.dirname(os.path.dirname(os.path.abspath(__file__))))

from db import repo
from services.auth_service import hash_password
from services.course_service import add_course, get_all_courses

DEFAULT_USERS = [
    {"username": "admin",   "password": "admin123",   "name": "Admin",   "role": "admin"},
    {"username": "teacher", "password": "teacher123", "name": "Teacher", "role": "teacher"},
]

DEFAULT_COURSES = [
    {"code": "Python",   "name": "Python Programming",      "duration": 1, "semesters": 2},
    {"code": "WebDev",   "name": "Web Development",         "duration": 1, "semesters": 2},
    {"code": "DataSci",  "name": "Data Science",            "duration": 1, "semesters": 2},
    {"code": "ML",       "name": "Machine Learning",        "duration": 1, "semesters": 2},
    {"code": "Java",     "name": "Java Programming",        "duration": 1, "semesters": 2},
    {"code": "Android",  "name": "Android Development",     "duration": 1, "semesters": 2},
    {"code": "UIUX",     "name": "UI/UX Design",            "duration": 1, "semesters": 2},
    {"code": "CySec",    "name": "Cybersecurity Essentials","duration": 1, "semesters": 2},
]


def seed():
    print("Seeding database...")

    # Users
    for u in DEFAULT_USERS:
        existing = repo.get_item(f"USER#{u['username']}", "#META")
        if not existing:
            repo.put_item({
                "PK": f"USER#{u['username']}", "SK": "#META",
                "GSI1_PK": "ALL#USERS", "GSI1_SK": f"USER#{u['username']}",
                "username": u["username"],
                "password_hash": hash_password(u["password"]),
                "name": u["name"],
                "role": u["role"]
            })
            print(f"  Created user: {u['username']} / {u['password']}")
        else:
            print(f"  User {u['username']} already exists")

    # Courses
    existing_codes = [c["code"] for c in get_all_courses()]
    for c in DEFAULT_COURSES:
        if c["code"] not in existing_codes:
            add_course(c["code"], c["name"], c["duration"], c["semesters"])
            print(f"  Created course: {c['code']} -- {c['name']}")
        else:
            print(f"  Course {c['code']} already exists")

    print("\nSeeding complete!")
    print("   admin   / admin123")
    print("   teacher / teacher123")


if __name__ == "__main__":
    seed()

File: _legacy/api/setup/seed.php
<?php
require_once __DIR__ . '/../config.php';
require_once __DIR__ . '/../helpers/db.php';

$db = get_db();

// Users
$users = [
    ['admin',   'admin123',   'Admin',   'admin'],
    ['teacher', 'teacher123', 'Teacher', 'teacher'],
];
foreach ($users as [$u, $p, $n, $r]) {
    $hash = hash('sha256', $p);
    $db->prepare('INSERT IGNORE INTO users (username, password_hash, password_plain, name, role) VALUES (?,?,?,?,?)')
       ->execute([$u, $hash, $p, $n, $r]);
    echo "  Created user: $u / $p\n";
}

// Courses — monthly_fee = per month, yearly_fee = per year
$courses = [
    ['PYTHON',  'Python Programming',        1, 3000, 30000],
    ['WEBDEV',  'Web Development',           1, 3500, 35000],
    ['DATASCI', 'Data Science',              1, 4000, 40000],
    ['ML',      'Machine Learning',          1, 4500, 45000],
    ['JAVA',    'Java Programming',          1, 3000, 30000],
    ['ANDROID', 'Android Development',       1, 3500, 35000],
    ['UIUX',    'UI/UX Design',              1, 2500, 25000],
    ['CYSEC',   'Cybersecurity Essentials',  1, 4000, 40000],
];
foreach ($courses as [$code, $name, $dur, $mfee, $yfee]) {
    $db->prepare('INSERT IGNORE INTO courses (code, name, duration, monthly_fee, yearly_fee) VALUES (?,?,?,?,?)')
       ->execute([$code, $name, $dur, $mfee, $yfee]);
    echo "  Created course: $code — $name (₹$mfee/mo · ₹$yfee/yr)\n";
}

echo "\nSeeding complete!\n";
echo "  Login: admin / admin123\n";
echo "  Login: teacher / teacher123\n";
